File: nagasena/test/org/openexi/proc/common/EXIOptionsTest.java
package org.openexi.proc.common;

import junit.framework.Assert;
import junit.framework.TestCase;

public class EXIOptionsTest extends TestCase {
  
  public EXIOptionsTest(String name) {
    super(name);
  }

  /**
   * <common> element will be output when getOutput is called
   * with outputSchemaId of value true and an schemaId is available.
   */
  public void testSchemaId_01() throws Exception {
    EXIOptions options = new EXIOptions();
    options.setSchemaId(new SchemaId(""));
    int outline = options.getOutline(true);
    Assert.assertTrue((outline & EXIOptions.ADD_COMMON) != 0);
  }

  /**
   * <common> element will *not* be output when getOutput is called
   * with outputSchemaId of value false even though an schemaId is
   * available.
   */
  public void testSchemaId_02() throws Exception {
    EXIOptions options = new EXIOptions();
    options.setSchemaId(new SchemaId(""));
    int outline = options.getOutline(false);
    Assert.assertTrue((outline & EXIOptions.ADD_COMMON) == 0);
  }

}

File: nagasenacs/test/Nagasena/Proc/Common/EXIOptionsTest.cs
using System;
using NUnit.Framework;

namespace Nagasena.Proc.Common {

  [TestFixture]
  public class EXIOptionsTest : Nagasena.LocaleLauncher {

    /// <summary>
    /// <common> element will be output when getOutput is called
    /// with outputSchemaId of value true and an schemaId is available.
    /// </summary>
    [Test]
    public virtual void testSchemaId_01() {
      EXIOptions options = new EXIOptions();
      options.SchemaId = new SchemaId("");
      int outline = options.getOutline(true);
      Assert.IsTrue((outline & EXIOptions.ADD_COMMON) != 0);
    }

    /// <summary>
    /// <common> element will *not* be output when getOutput is called
    /// with outputSchemaId of value false even though an schemaId is
    /// available.
    /// </summary>
    [Test]
    public virtual void testSchemaId_02() {
      EXIOptions options = new EXIOptions();
      options.SchemaId = new SchemaId("");
      int outline = options.getOutline(false);
      Assert.IsTrue((outline & EXIOptions.ADD_COMMON) == 0);
    }

  }

}